### PR TITLE
Releasing Bitext Extraction code

### PR DESCRIPTION
**Pull request by ferhanture**
ivory/lsh/bitext/FindParallelSentencePairs corresponds to the main algorithm, and ivory/lsh/bitext/FilterSentencePairs implements the second step (filtering) of the 2-step classification approach. Tested on German-English Wikipedia.

### CODE REVIEW

File: src/java/main/ivory/core/tokenize/LuceneArabicAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.ar.ArabicAnalyzer;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+import org.apache.lucene.util.Version;
+
+public class LuceneArabicAnalyzer extends Tokenizer {
+  ArabicAnalyzer a;
```
Comment (lintool) [inline src/java/main/ivory/core/tokenize/LuceneArabicAnalyzer.java]: private? better name?

File: src/java/main/ivory/core/tokenize/LuceneArabicAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        tokens.add(termAtt.toString());
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    String[] arr = new String[tokens.size()];
+    return tokens.toArray(arr);
+  }
+  
+  public static void main(String[] args) throws IOException, ClassNotFoundException, InstantiationException, IllegalAccessException{
```
Comment (lintool) [inline src/java/main/ivory/core/tokenize/LuceneArabicAnalyzer.java:51]: Please add test case?

File: src/java/main/ivory/lsh/bitext/FilterSentencePairs.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+	}
+
+	//	Map: (eSent, fSent) --> (eSent, fSent)
+	//	Ê Ê Ê(vect1, vect2) = convert sentences into tf-idf vectors
+	//	Ê Ê Êcompute features for vector pair
+	//		 emit pair if confidence met
+	private static class MyMapper extends MapReduceBase implements
+	Mapper<LongWritable, Text, Text, Text> {
+
+		PreprocessHelper helper;
```
Comment (lintool) [inline src/java/main/ivory/lsh/bitext/FilterSentencePairs.java]: private?

Comment (ferhanture) [inline src/java/main/ivory/lsh/bitext/FilterSentencePairs.java]: this applies to other variables as well, right? I don't directly reference any of them, so better keep them private?

File: src/java/main/ivory/core/tokenize/LuceneArabicAnalyzer.java (hunk 2 above)
Comment (ferhanture) [inline src/java/main/ivory/core/tokenize/LuceneArabicAnalyzer.java:51]: okay, working on it...

File: src/java/main/ivory/lsh/bitext/FilterSentencePairs.java (hunk 3 above)
Comment (lintool) [inline src/java/main/ivory/lsh/bitext/FilterSentencePairs.java]: Yes, always private explicitly unless you have a compelling reason 
otherwise.

**Pull request merged**